parlai/mturk/core/mturk_manager.py

```diff
@@ -290,7 +291,10 @@ def _handle_partner_disconnect(self, worker_id, assignment_id):
         elif not agent.state.is_final():
             # Update the assignment state
             agent.some_agent_disconnected = True
-            agent.state.status = AssignState.STATUS_PARTNER_DISCONNECT
+            if len(agent.state.messages) < self.minimum_messages:
```

Comment:
Technicality here - messages will include both messages sent by the worker and their partner. You should parse messages and only count the messages that have `message_data.id == agent.id` before making this.

Admittedly this seems confusing, as `message_data.id` is actually storing the sender id of the message, but that's my bad for some poor variable naming. Packets have an id and a data field, and only the data field is stored in messages.